common/persistence/client/fx.go

```diff
@@ -97,29 +97,36 @@ func EventBlobCacheProvider(
 func FactoryProvider(
 	params NewFactoryParams,
 ) Factory {
-	var requestRatelimiter quotas.RequestRateLimiter
+	var systemRequestRateLimiter, namespaceRequestRateLimiter quotas.RequestRateLimiter
 	if params.PersistenceMaxQPS != nil && params.PersistenceMaxQPS() > 0 {
 		if params.EnablePriorityRateLimiting != nil && params.EnablePriorityRateLimiting() {
-			requestRatelimiter = NewPriorityRateLimiter(
-				params.PersistenceNamespaceMaxQPS,
+			systemRequestRateLimiter = NewPriorityRateLimiter(
 				params.PersistenceMaxQPS,
-				params.PersistencePerShardNamespaceMaxQPS,
 				RequestPriorityFn,
 				params.OperatorRPSRatio,
 				params.HealthSignals,
 				params.DynamicRateLimitingParams,
 				params.MetricsHandler,
 				params.Logger,
 			)
+			namespaceRequestRateLimiter = NewPriorityNamespaceRateLimiter(
+				params.PersistenceMaxQPS,
+				params.PersistenceNamespaceMaxQPS,
+				params.PersistencePerShardNamespaceMaxQPS,
+				RequestPriorityFn,
+				params.OperatorRPSRatio,
+			)
 		} else {
-			requestRatelimiter = NewNoopPriorityRateLimiter(params.PersistenceMaxQPS)
+			systemRequestRateLimiter = NewNoopPriorityRateLimiter(params.PersistenceMaxQPS)
+			namespaceRequestRateLimiter = NewNoopPriorityRateLimiter(params.PersistenceMaxQPS)
```

Comment:
`params.PersistenceNamespaceMaxQPS` ? 

Actually it doesn't matter too much. We need to deprecate the EnablePriorityRateLimiting flag, it should always be true. All the new work we did is also not available when it's disabled.